combat-trainer.lic

```diff
@@ -2584,14 +2609,24 @@ class AttackProcess
   private
 
   def check_face(game_state)
-    return unless Flags['ct-face-what']
-
-    fput("eng #{game_state.npcs.first}")
-    Flags.reset('ct-face-what')
+    return unless Flags['ct-face-what'] || (DRRoom.dead_npcs.size > 0 && DRStats.empath?)
+    if (game_state.npcs.length > 0)
+      case bput("eng #{game_state.npcs.first}", 'You will have to retreat from your current melee first', 'You are already at melee', 'You begin to advance', 'You spin around to face', 'is already quite dead')
+      when /You will have to retreat from your current melee first/
+        fput("face #{game_state.npcs.first}")
+      when /is already quite dead/
+        case bput("eng #{game_state.npcs.last}", 'You will have to retreat from your current melee first', 'You are already at melee', 'You begin to advance', 'You spin around to face', 'is already quite dead')
+        when /You will have to retreat from your current melee first/
+          fput("face #{game_state.npcs.last}")
+        end
+      end
+      Flags.reset('ct-face-what')
+    end
   end
 
   def attack_melee(charged_maneuver, game_state)
     waitrt?
+    return if DRRoom.dead_npcs.size > 0 && DRStats.empath?
```

Comment:
What is the purpose of these returns throughout the actions? This seems like an issue that would only arise with the Empath attacking with weapons while also manipulating? When does this actually come up? I think a lot of these changes are being put through to support hunting manipulatable critters with non. This is a fairly uncommon issue I found only once (and that's where you mentioned with the reavers and hounds). Not sure it's worth always checking this for a one-off?  

Also, this could be `return if DRRoom.dead_npcs.empty? && DRStats.empath?` which seems clearer to me when reading it using the method, but either way is logically the same.

Reply:
FWIW, I was going to do this when, but then realized it only occurs once while hunting, so never seemed an issue worth fixing. I don't think this is a bad fix though and as long as it doesn't add terrible overhead, is probably not a big deal.

Reply:
The returns are to help prevent shock when hunting in undead/constructs with live npcs around.
If a creature dies and you try do an attack before getting to loot the dead creatures, you turn to face the closest creature, which is sometimes a live npc. And you get stunned and a bit of shock as an empath because of it.

Reply:
Yeah, but this is a check every single action for every single characters based on something that really only occurs once in the entire ladder, and you really only hunt these two critters for a short amount of time. Maybe a day? I just think I need more feedback from others, I only experienced this once, in the same place you did, from circle 0-165, and that's when I sold.

Reply:
OK, this needs to be handled differently, you can probably add the return of this to a single place within AttackProcess rather than every method inside. This would be a better solution.